### PR TITLE
StandaloneMmPkg : Add MM core fv location PPI

### PR DESCRIPTION
**Pull request by hongbin123**
- [ ] Breaking change?
  - **Breaking change** - Does this PR cause a break in build or boot behavior?
  - Examples: Does it add a new library class or move a module to a different repo.
- [ ] Impacts security?
  - **Security** - Does this PR have a direct security impact?
  - Examples: Crypto algorithm change or buffer overflow fix.
- [ ] Includes tests?
  - **Tests** - Does this PR include any explicit test code?
  - Examples: Unit tests or integration tests.

### CODE REVIEW

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c [hunk 1]
```diff
@@ -147,12 +147,41 @@ LocateMmCoreFv (
...
-  EFI_FV_INFO          VolumeInfo;
+  EFI_STATUS               Status;
+  UINTN                    FvIndex;
+  EFI_PEI_FV_HANDLE        VolumeHandle;
+  EFI_PEI_FILE_HANDLE      FileHandle;
+  EFI_PE32_SECTION         *SectionData;
+  EFI_FV_INFO              VolumeInfo;
+  MM_CORE_FV_LOCATION_PPI  *MmCoreFvLocation;
+
+  Status = PeiServicesLocatePpi (&gMmCoreFvLocationPpiGuid, 0, NULL, (VOID **)&MmCoreFvLocation);
```
Comment (niruiyu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c:158]: Add comments to explain that the code below is to use FV indicated by MmCoreFvLocation PPI directly.
And clarify that it's the producer's responsibility that MmCoreFvLocation PPI reports the correct FV that contains MmCore.

Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c:158]: ok

File: StandaloneMmPkg/Include/Ppi/MmCoreFvLocationPpi.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/** @file
+  MM Core FV location PPI header file.
+
+  MM Core FV location PPI is used by StandaloneMm IPL to find MM Core
+  .
```
Comment (jiaxinwu) [inline StandaloneMmPkg/Include/Ppi/MmCoreFvLocationPpi.h:5]: MM Core FV location PPI is used by StandaloneMm IPL to find MM Core.

"." is in another line

**Comment by jiaxinwu** [comment]
"MmIpl could locate MM core FV location PPI to find current MM
FV location."

--> 

"MmIpl should locate MM core FV location PPI to find current MM
FV location."

File: StandaloneMmPkg/Include/Ppi/MmCoreFvLocationPpi.h (hunk 2 above)
Comment (hongbin123) [reply StandaloneMmPkg/Include/Ppi/MmCoreFvLocationPpi.h:5]: updated

**Comment by hongbin123** [comment]
> "MmIpl could locate MM core FV location PPI to find current MM FV location."
> 
> -->
> 
> "MmIpl should locate MM core FV location PPI to find current MM FV location."

updated

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c [hunk 3]
```diff
@@ -147,12 +147,45 @@ LocateMmCoreFv (
...
+  EFI_STATUS               Status;
+  UINTN                    FvIndex;
+  EFI_PEI_FV_HANDLE        VolumeHandle;
+  EFI_PEI_FILE_HANDLE      FileHandle;
+  EFI_PE32_SECTION         *SectionData;
+  EFI_FV_INFO              VolumeInfo;
+  MM_CORE_FV_LOCATION_PPI  *MmCoreFvLocation;
+
+  //
+  // FSP-I FV or bootloader MM FV will install MM Core FV location PPI
```
[the remark this answers is not in the document]
Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c]: updated

File: StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c [hunk 4]
```diff
@@ -147,12 +147,44 @@ LocateMmCoreFv (
...
+  EFI_STATUS               Status;
+  UINTN                    FvIndex;
+  EFI_PEI_FV_HANDLE        VolumeHandle;
+  EFI_PEI_FILE_HANDLE      FileHandle;
+  EFI_PE32_SECTION         *SectionData;
+  EFI_FV_INFO              VolumeInfo;
+  MM_CORE_FV_LOCATION_PPI  *MmCoreFvLocation;
+
+  //
+  // MmIpl should locate MM Core FV location PPI to find the MM core FV.
```
Comment (jiaxinwu) [inline StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c:159]: Ray asked to clarify that it's the producer's responsibility that MmCoreFvLocation PPI reports the correct FV that contains MmCore.
And also clarify what's the behavior if not found the gMmCoreFvLocationPpiGuid:

The producer of the MmCoreFvLocation PPI is responsible for ensuring that it reports the correct Firmware Volume (FV) containing the MmCore.
If the gMmCoreFvLocationPpiGuid is not found, the system will search all Firmware Volumes (FVs) to locate the FV that contains the MM Core.

Comment (hongbin123) [reply StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c]: updated

**Review by niruiyu — APPROVED** [review]

**Pull request merged**